File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.swift
class Solution {
    func topKFrequent(_ nums: [Int], _ k: Int) -> [Int] {
        var res: [Int: Int] = [:]
        for num in nums {
            res[num, default: 0] += 1
        }

        // 2. Sort by frequency (descending) and take top k keys
        let sortedItems = res.sorted { $0.value > $1.value }
        return sortedItems.prefix(k).map { $0.key }
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.py
class Solution:
    #[3,4,5,6,1,2]
    # l
    #     m
    #           r
    def findMin(self, nums: List[int]) -> int:
        s,e = 0, len(nums)-1
        while s < e:
            m = (s+e)//2
            if nums[m] > nums[e]:
                s = m + 1
            else:
                e = m
        return nums[s]